wordpress-mobile/WordPress-iOS
# Adjust i18n issues for Page Design and Site Design flows

## Description

**Related PR:** https://github.com/wordpress-mobile/WordPressKit-iOS/pull/393

In #15231 we added `language` as a parameter to the site-design endpoint based on the current server-side specifications. That has been carried over with our other changes to this endpoint. 

This PR does three things:
1. Updates the site design flow to use the v2 locale key `_locale`
2. Updates the page design flows to use the v2 locale key `_locale`
3. Corrects an issue on the Site Design Preview screen where the `Choose` button wasn't translated.

## To test:

> Updates the site design flow to use the v2 locale key `_locale`

This is easiest to test with a Proxy tool like [Charles Proxy ](https://www.charlesproxy.com/) since the backend has made the necessary changes to support our live apps.

1. Set up your proxy service of choice
2. Set your device language to be something other than English. I tested with US Spanish (slug = `es`)
3. Navigate to site design screen in site creation
4. Expect to see a request similar to:
`https://public-api.wordpress.com/wpcom/v2/common-starter-site-designs?_locale=es&....{{remaining_request_parameters}}... `
    - Notice: `_locale=es` where the `_locale` key should be populated with your selected language's slug.
    - Notice: `language` and `locale` should not be present

> Updates the page design flows to use the v2 locale key `_locale`

This is easiest to test with a Proxy tool like [Charles Proxy ](https://www.charlesproxy.com/) since the backend has made the necessary changes to support our live apps.

1. Set up your proxy service of choice
2. Set your device language to be something other than English. I tested with US Spanish (slug = `es`)
3. Navigate to page design screen in the page creation flow for a dotCom site.
4. Expect to see a request similar to:
`https://public-api.wordpress.com/wpcom/v2/sites/176997822/block-layouts?_locale=es&....{{remaining_request_parameters}}...`
    - Notice: `_locale=es` where the `_locale` key should be populated with your selected language's slug.
 5. Regeat for a Self-hosted site
     - 📓 The request will now be made to `https://public-api.wordpress.com/wpcom/v2/common-block-layouts?_locale=es`

> Corrects an issue on the Site Design Preview screen where the `Choose` button wasn't translated.

1. Set your device language to be something other than English. I tested with US Spanish
2. Navigate to site design screen in site creation
3. Select a Design
4. Select the Preview Button (LHS)
5. Expect the `Choose` Button to be Translated

<img width=300 src="https://user-images.githubusercontent.com/3384451/116563508-17279280-a8d2-11eb-8881-fa5cdb89ef0c.PNG" />
 
## Regression Notes
1. Potential unintended areas of impact
None outside of the changes mentioned here.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

## PR submission checklist:
